db/system_keyspace.cc

```diff
@@ -262,6 +262,24 @@ schema_ptr raft_snapshots() {
     return schema;
 }
 
+schema_ptr repair() {
+    static thread_local auto schema = [] {
+        auto id = generate_legacy_id(NAME, REPAIR);
+        return schema_builder(NAME, REPAIR, std::optional(id))
+            .with_column("table_uuid", uuid_type, column_kind::partition_key)
+            .with_column("range_start", utf8_type, column_kind::partition_key)
+            .with_column("range_end", utf8_type, column_kind::partition_key)
+            .with_column("repair_uuid", uuid_type, column_kind::clustering_key)
+            .with_column("repair_time", timestamp_type, column_kind::clustering_key)
```

Comment:
Why are these clustering keys? Isn't it enough to keep the last repair for a range?

Reply:
I want this table to contain the full repair history. Since this table is read once during boot up only, having more entries will not make runtime slower.

Reply:
Having repair history is a good idea, but then we should have a TTL (say 1 year) so the data has some bounds.

The time should be the first column so repairs are sorted by time. Maybe we can change the repair uuid to be of timeuuid type, and so we can have just the repair_uuid column.

Reply:
This sounds a good idea.

Reply:
Ok, please send a separate patch, so it can be ready for this larger change.

Reply:
ok.

Reply:
AFAIK we already have tables to manage token ranges in distributed manner - and keep track of whats going on. The fact a range has been repaired is a global (or dc local) attribute not a node local attribute. 

The advantages of managing this globally are numerous - we will not need to deal with topology changes and can reuse the knowledge of data being repaired across nodes 

We are doing this in MVs in some form.